via/static/scripts/video_player/hooks/use-side-by-side-layout.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useEffect, useState } from 'preact/hooks';
+
+type SidebarLayout = {
+  /** Whether sidebar is open or closed */
+  expanded: boolean;
+  /** Current width of sidebar in pixels */
+  width: number;
+  /** Current height of sidebar in pixels */
+  height: number;
+  /** Width of controls (toolbar, bucket bar) on the edge of the sidebar */
+  toolbarWidth: number;
+};
+
+type LayoutChangeEventDetail = {
+  sideBySideActive: boolean;
+  sidebarLayout: SidebarLayout;
+};
+
+type LayoutChangeEvent = CustomEvent & { detail: LayoutChangeEventDetail };
+
+// Minimum body width at which laying out side-by-side should be considered
+export const SIDEBYSIDE_THRESHOLD = 875;
+// The sidebar reports its toolbar width as 33px. This is a custom padding
+// amount that allows this app to nestle up more cloesly to the sidebar when
+// closed.
+export const TOOLBAR_WIDTH = 22;
+
+/**
+ * Adjust the available space to the right of app content to make room for
+ * the sidebar when plausible. This mimics the client's "side-by-side" mode,
+ * tuned specifically for this application.
+ *
+ * Side-by-side layout will be re-computed whenever sidebar state via the
+ * `hypothesis:layoutchange` event changes meaningfully, or when the window
+ * is resized.
+ */
+export function useSideBySideLayout() {
+  const [sideBySideApplicable, setSideBySideApplicable] = useState(false);
+  const [sufficientSpace, setSufficientSpace] = useState(
+    document.body.clientWidth >= SIDEBYSIDE_THRESHOLD
+  );
+  const [sidebarWidth, setSidebarWidth] = useState(0);
+
+  const updateSidebarState = useCallback((e: Event) => {
+    const sidebarInfo: LayoutChangeEventDetail = (e as LayoutChangeEvent)
+      .detail;
+    const sideBySide = sidebarInfo.sideBySideActive;
```

Comment:
I was attempting to decouple this hook from the logic that sets the config (disables the side-by-side mode)—this hook can't be certain that that config has been set. Is there any _harm_ in having this check here? I could at least comment it better.

Reply:
> Is there any harm in having this check here?

Yes. It will break when https://github.com/hypothesis/client/pull/5623 lands. Some ways we could resolve that:

- Have the caller of this hook specify the side-by-side mode
- Just document the assumption that manual side-by-side mode is being used
- Add a field to the `hypothesis:layoutchange` event to indicate the mode